DataStructures/ArrayBasedCircularQueue/Enumerators/QueueEnumerator.cs

```diff
@@ -0,0 +1,100 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Text;
+
+namespace DataStructures.ArrayBasedCircularQueue.Enumerators
+{
+    /// <summary>
+    /// Queue Enumerator to enumerate queue.
+    /// </summary>
+    /// <typeparam name="T">Generic Type.</typeparam>
+    public class QueueEnumerator<T> : IEnumerator<T>
+    {
+        private readonly T[] queue;
+        private bool disposedValue = false;
+        private int front = 0;
+        private int rear = 0;
```

Comment:
Codacy found an issue: [Remove this initialization to 'rear', the compiler will do that for you.](https://app.codacy.com/manual/siriak/C-Sharp/pullRequest?prid=6301193)